examples/graphbolt/download.py

```diff
@@ -0,0 +1,597 @@
+        "paper": 269346174,
+    },
+    "large": {
+        "paper": 100000000,
+    },
+    "medium": {
+        "paper": 10000000,
+    },
+    "small": {
+        "paper": 1000000,
+    },
+    "tiny": {
+        "paper": 100000,
+    },
+}
+
+num_edges = {
+    "full": {
+        "paper__cites__paper": 3996442004,
+    },
+    "large": {
+        "paper__cites__paper": 1223571364,
+    },
+    "medium": {
+        "paper__cites__paper": 120077694,
+    },
+    "small": {
+        "paper__cites__paper": 12070502,
+    },
+    "tiny": {
+        "paper__cites__paper": 447416,
+    },
+}
+
+
+def split_data(label_path, set_dir, dataset_size, class_num):
+    """This is for splitting the labels into three sets: train, validation, and test sets."""
+    # labels = np.memmap(label_path, dtype='int32', mode='r',  shape=(num_nodes[dataset_size]["paper"], 1))
+    labels = np.load(label_path)
+
+    total_samples = len(labels)
+    train_end = int(0.6 * total_samples)
+    validation_end = int(0.8 * total_samples)
+
+    indices = np.arange(total_samples)
+    train_indices = indices[:train_end]
+    validation_indices = indices[train_end:validation_end]
+    test_indices = indices[validation_end:]
+    print(indices)
+    print(train_indices)
+    print(validation_indices)
+    print(test_indices)
+
+    train_labels = labels[:train_end].astype(np.int64)
+    validation_labels = labels[train_end:validation_end].astype(np.int64)
+    test_labels = labels[validation_end:].astype(np.int64)
+    print(train_labels, len(train_labels))
+    print(validation_labels, len(validation_labels))
+    print(test_labels, len(test_labels))
+
+    gb.numpy_save_aligned(
+        f"{set_dir}/train_indices_{class_num}.npy", train_indices
+    )
+    gb.numpy_save_aligned(
+        f"{set_dir}/validation_indices_{class_num}.npy", validation_indices
+    )
+    gb.numpy_save_aligned(
+        f"{set_dir}/test_indices_{class_num}.npy", test_indices
+    )
+    gb.numpy_save_aligned(
+        f"{set_dir}/train_labels_{class_num}.npy", train_labels
+    )
+    gb.numpy_save_aligned(
+        f"{set_dir}/validation_labels_{class_num}.npy", validation_labels
+    )
+    gb.numpy_save_aligned(f"{set_dir}/test_labels_{class_num}.npy", test_labels)
+
+
+def add_edges(edges, source, dest, dataset_size):
+    """This is for processing the edges in the graph and convert them to correct shape."""
+    for edge in edges:
+        print(f"\t Processing {edge} edge...")
+
+        old_edge_path = source + "/" + edge + "/" + "edge_index.npy"
+        new_edge_path = dest + "/" + edge + ".npy"
+        rev_edge_path = dest + "/" + edge + "_rev.npy"
+        os.rename(src=old_edge_path, dst=new_edge_path)
+
+        # edge_array = np.memmap(new_edge_path, dtype='int32', mode='r',  shape=(num_edges[dataset_size][edge], 2))
+        edge_array = np.load(new_edge_path)
+        new_edge_array = edge_array.transpose()
+        rev_edge_array = np.ascontiguousarray(new_edge_array[::-1])
```

Comment:
Can we remove self edges from new_edge_array, then concatenate rev_edge_array to it and then add back the self edges? We don't need 2 edge types for the homo dataset, we only need 1 edge type.

Reply:
Sure.

Reply:
Then we can test whether adding the reverse edges helps or hurts the accuracy for the homo case.

Reply:
I see, I will put one paper__cites__paper file with the normal and reverse concatenated

Reply:
Let's do this (to clarify):
Remove self edges
concatenate reverse edges
add self edges

Reply:
I tried on the tiny dataset but the 2983 classes result still is lower:
19 class: 73.00%
2983 classes: 66.22%

Reply:
Can we try with the medium? If it does not work, we can add the dataset without the reverse edges.

Reply:
And can you show me the final code?

Reply:
Actually, the HOM dataset in the paper also have acc drop in 2K classes task comparing to the 19 class task, which looks normal and our result looks even better. However, the HET dataset they have the 2K classes tasks comparable or even better than 19 classes task, which does not make sense to me.

Reply:
Can you show results with reverse edges, without reverse edges for 19 and 2K tasks?

Reply:
Sure. I put the code in the Slack